code/modules/research/experimentor/loot/device_exploot.dm

```diff
@@ -0,0 +1,236 @@
+			explosion(src.loc, -1, rand(potency/50,potency/30), rand(potency/30,potency/20), rand(potency/20,potency/15), flame_range = potency/25)
+			warn_admins(user, "Explosion")
+			qdel(src)
+
+// Cleaner
+/obj/item/discovered_tech/cleaner/initialize()
+	..()
+	origin_tech += "engineering=[1+raritylevel+rand(raritylevel,raritylevel+1)]"
+	keywords = list("destruction")
+
+/obj/item/discovered_tech/cleaner/itemproc(mob/user)
+	playsound(src.loc, "sparks", rand(25,50), 1)
+	var/obj/item/grenade/chem_grenade/cleaner/CL = new/obj/item/grenade/chem_grenade/cleaner(get_turf(user))
+	CL.prime()
+	warn_admins(user, "Cleaning Foam", 0)
+
+// Flashbang
+/obj/item/discovered_tech/flashbang/initialize()
+	..()
+	origin_tech += "plasmatech=[1+raritylevel+rand(raritylevel,raritylevel+1)];"
+	keywords = list("light", "sound")
+
+/obj/item/discovered_tech/flashbang/itemproc(mob/user)
+	playsound(src.loc, "sparks", rand(25,50), 1)
+	var/obj/item/grenade/flashbang/CB = new/obj/item/grenade/flashbang(get_turf(user))
+	CB.range = rand(stability/25, max(potency/10, 4))
+	CB.prime()
+	warn_admins(user, "Flashbang")
+
+// Rapid Duplicator
+/obj/item/discovered_tech/rapidDuplicator/initialize()
+	..()
+	origin_tech += "magnets=[1+raritylevel+rand(raritylevel,raritylevel+1)];"
+	origin_tech += "engineering=[1+raritylevel+rand(raritylevel,raritylevel+1)]"
+	origin_tech += "materials=[1+raritylevel+rand(raritylevel,raritylevel+1)];"
+	keywords = list("replication")
+
+/obj/item/discovered_tech/rapidDuplicator/itemproc(mob/user)
+	if(raritylevel >= RARITY_UNCOMMON)
+		audible_message("[src] emits a loud pop and sprays dangerous projectiles everywhere!")
+		warn_admins(user, "Dangerous Rapid Duplicator")
+	else
+		audible_message("[src] emits a loud pop!")
+		warn_admins(user, "Rapid Duplicator", 0)
+	var/list/dupes = list()
+	var/counter
+	var/max = rand(stability/20, max(potency/5, 5))
+	for(counter = 1; counter < max; counter++)
+		var/obj/item/discovered_tech/R = new src.type(get_turf(src))
+		R.icon_state = icon_state
+		R.name = name
+		R.desc = desc
+		R.throwforce = ((potency/10)*raritylevel)+1
+		dupes |= R
+		spawn()
+			R.throw_at(pick(oview(4+potency/20,get_turf(src))),10,1)
+	counter = 0
+	spawn(rand(stability/10,stability))
+		for(counter = 1; counter <= dupes.len; counter++)
+			var/obj/item/discovered_tech/R = dupes[counter]
+			qdel(R)
+
+// Teleporter
+/obj/item/discovered_tech/teleport/initialize()
+	..()
+	origin_tech += "bluespace=[1+raritylevel+rand(raritylevel,raritylevel+1)];"
+	origin_tech += "powerstorage=[1+raritylevel+rand(raritylevel,raritylevel+1)]"
+	keywords = list("teleportation", "bluespace")
+
+/obj/item/discovered_tech/teleport/itemproc(mob/user)
+	to_chat(user, "<span class='notice'>The [src] begins to vibrate!</span>")
+	spawn(rand((15-stability/10),30-stability/5))
+		var/turf/userturf = get_turf(user)
+		if(src.loc != user || is_teleport_allowed(userturf.z) == FALSE)
```

Comment:
== FALSE can be placed with `!statement` so in this case `!is_teleport_allowed(userturf.z)`